stdlib/public/core/Optional.swift

```diff
@@ -340,6 +440,26 @@ public func ~= <T>(lhs: _OptionalNilComparisonType, rhs: T?) -> Bool {
 
 // Enable equality comparisons against the nil literal, even if the
 // element type isn't equatable
+
+/// Returns a Boolean value indicating whether the left-hand-side argument is
+/// `nil`.
+///
+/// You can use this equal-to operator (`==`) to test whether an optional
+/// instance is `nil` even when the wrapped value's type does not conform to
+/// the `Equatable` protocol. The following example declares the `numbers`
+/// variable as an optional array of integers. Although `Array<Int>` is not an
+/// `Equatable` type, this operator allows checking whether `numbers` is
+/// `nil`.
+///
+///     var numbers: [Int]?
+///     if numbers == nil {
+///         print("No numbers to be found.")
```

Comment:
Using an optional array in an example also makes me nervous because of the whole empty-vs-nil thing. Can we maybe use an optional scalar instead?